tests/test_full_jnc.py

```diff
@@ -0,0 +1,69 @@
+import sys
+import pytest
+import pexpect
+
+
+class JncTestBase:
+    def __init__(self, basedir):
+        self.px_args = dict(cwd=basedir, encoding="utf-8", logfile=sys.stdout)
+        self.basedir = basedir
+
+    def start_confd(self):
+        print('compile and start confd')
+        confd = pexpect.spawn('make clean all start', **self.px_args)
+        confd.expect(pexpect.EOF)
+        confd.close()
+        assert 0 == confd.exitstatus
+
+    def start_jnc(self):
+        print('compile and start the test')
+        self.jnc_runner = pexpect.spawn('gradle run', **self.px_args)
+
+    def cleanup(self):
+        self.jnc_runner.close()
+        confd = pexpect.spawn('make stop', **self.px_args)
+        confd.expect(pexpect.EOF)
+        confd.close()
+
+
+@pytest.fixture(scope='function',
+                params=['1-test-union',
```

Comment:
possibly make some automated ls/grep for directories with "test" (or similar) to save on this hidden constant maintenance?

Reply:
Is it worth it? I do not think it is likely we'll be adding more tests here; if we start doing that, maybe we shall rewrite this then.